CoViewingExample/ViewController.swift

```diff
@@ -0,0 +1,676 @@
+            // Room `name`, the Client will create one for you. You can get the name or sid from any connected Room.
+            builder.roomName = "twilio"
+
+            // Restrict video bandwidth used by viewers to improve presenter video. Use more bandwidth for presenter audio.
+            if name == "viewer" {
+                builder.encodingParameters = TVIEncodingParameters(audioBitrate: 0, videoBitrate: 1024 * 900)
+            } else {
+                builder.encodingParameters = TVIEncodingParameters(audioBitrate: 1024 * 96, videoBitrate: 0)
+            }
+        }
+
+        // Connect to the Room using the options we provided.
+        room = TwilioVideo.connect(with: connectOptions, delegate: self)
+        print("Attempting to connect to:", connectOptions.roomName as Any)
+
+        self.showRoomUI(inRoom: true)
+    }
+
+    func prepareLocalMedia() {
+        // All Participants share local audio and video when they connect to the Room.
+        // Create an audio track.
+        if (localAudioTrack == nil) {
+            localAudioTrack = TVILocalAudioTrack.init()
+
+            if (localAudioTrack == nil) {
+                print("Failed to create audio track")
+            }
+        }
+
+        // Create a camera video Track.
+        #if !targetEnvironment(simulator)
+        if (localVideoTrack == nil) {
+            // Preview our local camera track in the local video preview view.
+
+            camera = TVICameraCapturer(source: .frontCamera, delegate: nil)
+            let constraints = TVIVideoConstraints.init { (builder) in
+                builder.maxSize = TVIVideoConstraintsSize480x360
+                builder.maxFrameRate = TVIVideoConstraintsFrameRate24
+            }
+
+            localVideoTrack = TVILocalVideoTrack(capturer: camera!,
+                                                 enabled: true,
+                                                 constraints: constraints,
+                                                 name: "camera")
+            localVideoTrack.addRenderer(self.localView)
+            // We use the front facing camera only. Set mirroring each time since the renderer might be reused.
+            localView.shouldMirror = true
+        }
+        #endif
+    }
+
+    func showRoomUI(inRoom: Bool) {
+        self.hangupButton.isHidden = !inRoom
+        self.localView.isHidden = !inRoom
+        if (self.isPresenter == false) {
+            self.remotePlayerView.isHidden = !inRoom
+        }
+        self.remoteView.isHidden = !inRoom
+        self.presenterButton.isHidden = inRoom
+        self.viewerButton.isHidden = inRoom
+        self.setNeedsUpdateOfHomeIndicatorAutoHidden()
+        self.setNeedsStatusBarAppearanceUpdate()
+
+        if inRoom == false {
+            UIView.animate(withDuration: 0.2) {
+                self.view.backgroundColor = .white
+            }
+        }
+    }
```

Comment:
Ran into this crash once. ( I think I disconnect and then tried to connect to the same room? (I might have ran into invalid token.. not entirely sure though)
![screen shot 2018-11-09 at 4 33 46 pm](https://user-images.githubusercontent.com/17037593/48295336-d278ba00-e43f-11e8-9560-4d7d5f27063c.png)

Reply:
Yep, I've seen this one. We need to create `self.remotePlayerView` dynamically. 👍